### PR TITLE
gateway-api: improve secret sync resiliency

### PR DESCRIPTION
**Pull request by mhofstetter**
This PR improves the resiliency of the Gateway API secret sync.

main changes:
* delete synced secret if source secret gets deleted
* delete synced secret if source secret is no longer referenced by a Cilium Gateway
* trigger reconcile when the synced secret gets deleted or updated

Please review the individual commits.

This is a preparation to re-use the secret synchronization for Cilium Ingress (#28911)

### CODE REVIEW

**Comment by mhofstetter** [comment]
updated the test-fixtures

File: operator/pkg/gateway-api/secret.go [hunk 1]
```diff
@@ -92,3 +101,51 @@ func (r *secretSyncer) enqueueRequestForGatewayTLS() handler.EventHandler {
 		return reqs
 	})
 }
+
+func enqueueOwningSecretFromLabels() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+		requests := []reconcile.Request{}
+		labels := o.GetLabels()
+
+		if labels != nil {
```
Comment (meyskens) [inline operator/pkg/gateway-api/secret.go]: If i read this correct labels not being nil is the success path? Would it be a readibility improvement to do
```
if labels == nil {
  return
}
```

So the success path stays left aligned (i almost overread it)

**Review by meyskens — APPROVED** [review]
LGTM just one small thing for readability but giving you a green tick anyway :D

File: operator/pkg/gateway-api/secret.go (hunk 1 above)
Comment (mhofstetter) [reply operator/pkg/gateway-api/secret.go]: Thanks. fully agree. while on it, i also return if the namespace & name labels aren't present. This way the success path is fully left-aligned.

**Comment by mhofstetter** [comment]
Addressed @meyskens input - Thanks for the review!

**Review by sayboras — APPROVED** [review]
Thanks :heavy_check_mark:

**Pull request merged**